cmd/state/test_state.go

```diff
@@ -0,0 +1,155 @@
+package state
+
+import (
+	"bytes"
+	"context"
+	"io"
+	"net"
+	"os/signal"
+	"runtime"
+	"strconv"
+	"sync"
+	"sync/atomic"
+	"testing"
+
+	"github.com/sirupsen/logrus"
+	"github.com/spf13/afero"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"go.k6.io/k6/lib/testutils"
+	"go.k6.io/k6/ui/console"
+)
+
+// GlobalTestState is a wrapper around GlobalState for use in tests.
+type GlobalTestState struct {
+	*GlobalState
+	Cancel func()
+
+	Stdout, Stderr *bytes.Buffer
+	LoggerHook     *testutils.SimpleLogrusHook
+
+	Cwd string
+
+	ExpectedExitCode int
+}
+
+// NewGlobalTestState returns an initialized GlobalTestState, mocking all
+// GlobalState fields for use in tests.
+func NewGlobalTestState(t *testing.T) *GlobalTestState {
```

Comment:
However, since we import this from other packages, it can't be in a `something_test.go` file, right? So we probably need a separate package for it, similar to what we have done with `testutils` and `modulestest`, see https://github.com/grafana/k6/tree/master/lib/testutils and https://github.com/grafana/k6/tree/master/js/modulestest

Reply:
> Do you know how we could confirm this?

`go list -f '{{range $imp := .Imports}}{{printf "%s\n" $imp}}{{end}}' | sort` showed the `testing` package was imported and now it's not, but I didn't notice the binary size actually decrease after pulling your latest changes :thinking: `go version -m /path/to/k6/binary` doesn't actually show stdlib sub-packages were included :disappointed:

Reply:
not sure how the magic of https://github.com/jondot/goweight works and how accurate it is, but running it in the repo before and after your change actually shows a huge difference:
```
% diff pre.txt post.txt 
28d27
<   1.6 MB github.com/stretchr/testify/assert
32d30
<   1.3 MB testing
54d51
<   1.0 MB github.com/stretchr/testify/require
115d111
<   483 kB github.com/davecgh/go-spew/spew
158d153
<   319 kB net/http/httptest
194d188
<   222 kB github.com/pmezard/go-difflib/difflib
211d204
<   183 kB go.k6.io/k6/cmd/state
221d213
<   174 kB go.k6.io/k6/lib/testutils
267a260
>   115 kB go.k6.io/k6/cmd/state
310d302
<    75 kB runtime/trace
407d398
<    14 kB internal/sysinfo
414d404
<    10 kB net/http/internal/testcert
```

So, yeah, it seems like it was both a real problem and that it is now fixed :tada:

Reply:
Thanks, I was looking for something that could be run from the repo root to list imports of all sub-packages, as they would be assembled by `go build`.

The size does decrease by a few kBs:

```
42029102 k6-03f8cee3
42004497 k6-24717844
```

So maybe `testing` is not that heavy(?).

Reply:
Ah, cheers for mentioning `goweight`. :bookmark:

But it seems that `go build` strips out things like `testing` then?

Reply:
there is actually a tiny difference in binary size
```
41915384  k6-post
41941793  k6-pre
```

I have no idea why it's only 25kb though, maybe Go is really, really go at tree shaking :sweat_smile: though given the things you can do with reflection, I am actually leaning towards a more uncharitable interpretation - that we are already somehow including the `testing` package and `goweight` just doesn't detect it :thinking:

Reply:
I actually find it more interesting that our binaries differ by ~1MB! :thinking:

I had added #1996 to discuss for the upcoming cycle. It will never be a good time to tackle it, but the sooner we do it, the better.